### PR TITLE
feat(worker): Enforce limits at PR creation time

### LINKED ISSUE #7075: Enforce prXLimit values at PR creation
**What would you like Renovate to be able to do?**

Renovate should enforce prHourlyLimit and prConcurrentLimit at *PR creation time* too, not just at *branch creation time*.

**Did you already have any implementation ideas?**

Keep the existing checks at branch creation but then also make sure we are not creating PRs beyond the limits defined.

### PR DESCRIPTION
**Pull request by zharinov**
Closes #7075

If I understand it correctly, we should not consider Onboarding PRs here

### CODE REVIEW

File: lib/workers/branch/index.ts [hunk 1]
```diff
@@ -560,7 +560,10 @@ export async function processBranch(
     logger.debug(
       `There are ${config.errors.length} errors and ${config.warnings.length} warnings`
     );
-    const { prResult: result, pr } = await ensurePr(config);
+    const { prResult: result, pr } = await ensurePr(config, prLimitReached);
```
Comment (rarkins) [inline lib/workers/branch/index.ts:563]: Potential problem: isn't `prLimitReached` coming from the *branch* limit, instead of PR limit?We should make sure our logic is sound here

**Comment by rarkins** [comment]
We should think of ways to test this. e.g.
- Onboard a new repo with 2 PRs created
- Rename and close one, but leave the branch
- Reduce prConcurrentLimit to 1
- Run again and verify that we don't recreate the PR

File: lib/workers/branch/index.ts (hunk 1 above)
Comment (zharinov) [reply lib/workers/branch/index.ts:563]: Right, haven't noticed `getConcurrentPrsRemaining` isn't using any `platform` PR functions

Comment (zharinov) [reply lib/workers/branch/index.ts:563]: I'm about to rewrite it with `platform.findPr({ branchName })`, so both will be using `getPrList()` under the hood

**Comment by zharinov** [comment]
https://github.com/zharinov/renovate-test-limits/pulls

```
DEBUG: findPr(renovate/org.apache.lucene-lucene-queryparser-8.x, undefined, open) (repository=zharinov/renovate-test-limits)
DEBUG: 1 PRs are currently open (repository=zharinov/renovate-test-limits)
DEBUG: PR concurrent limit remaining: 0 (repository=zharinov/renovate-test-limits)
DEBUG: Calculated maximum PRs remaining this run (repository=zharinov/renovate-test-limits)
       "prsRemaining": 0

...
DEBUG: Creating PR (repository=zharinov/renovate-test-limits, branch=renovate/org.apache.lucene-lucene-queryparser-8.x)
       "prTitle": "Update dependency org.apache.lucene:lucene-queryparser to v8.6.1"
DEBUG: Reached PR limit - skipping branch creation (repository=zharinov/renovate-test-limits, branch=renovate/org.apache.lucene-lucene-queryparser-8.x)
```

File: lib/workers/branch/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/workers/branch/index.ts:563]: Both what? We ideally want the branch limits to check branches, and PR limits to check PRs.

Comment (zharinov) [reply lib/workers/branch/index.ts:563]: Okay, now I see (#7076)

File: lib/workers/repository/process/limits.ts [hunk 2]
```diff
@@ -48,7 +47,7 @@ export async function getConcurrentPrsRemaining(
     logger.debug(`Enforcing prConcurrentLimit (${config.prConcurrentLimit})`);
     let currentlyOpen = 0;
     for (const branch of branches) {
-      if (await branchExists(branch.branchName)) {
+      if (await platform.getBranchPr(branch.branchName)) {
```
Comment (zharinov) [inline lib/workers/repository/process/limits.ts:50]: Basically, it reduces function result, because branches can exist without associated PRs.

Comment (zharinov) [reply lib/workers/repository/process/limits.ts:50]: But as we don't count such branches, it means we potentially may create _more_ branches than before. Though, as we use same counter for now, corresponding PRs should be created too.

**Comment by rarkins** [comment]
@zharinov can you summarize what changes this PR is making? i.e. same way of calculating concurrent and hourly as before, but now enforcing the limits in two places instead of one? Or do you think this requires a change in calculations too?

**Comment by zharinov** [comment]
For the test scenario you proposed above, changes to calculations in `limits.ts` do not make any difference. Because of that, let's leave the as-is for now. Thus, in this PR we're only passing the `boolean` flag one level deeper (to `ensurePr`), where we make sure no new PR is created off the limit.
